src/NuGet.Clients/NuGet.Console/OutputConsole/OutputConsole.cs

```diff
@@ -71,8 +80,8 @@ public override async Task WriteAsync(string text)
 
             Start();
 
-            await NuGetUIThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
-            VsOutputWindowPane.OutputStringThreadSafe(text);
+            var outputWindowTextWriter = await _outputWindowTextWriter.GetValueAsync();
+            await outputWindowTextWriter.WriteAsync(text);
```

Comment:
Since this completes before actually "flushing" to the main thread (which is great, because it means faster logging and fewer main thread transitions), you should review your other methods on this class (e.g. `ClearAsync`) and add `await outputWindowTextWriter.FlushAsync()` before doing things that should apply to all prior write calls.